lib/money/money.rb

```diff
@@ -266,8 +270,10 @@ def all_rational?(splits)
   def value_to_decimal(num)
     if num.respond_to?(:to_d)
       num.is_a?(Rational) ? num.to_d(16) : num.to_d
-    else
+    elsif num.is_a?(Money) || num.is_a?(Numeric) || num.is_a?(String)
```

Comment:
In order for the "raises ArgumentError with unsupported argument" spec from RubyMoney to pass we can't just call `num#to_s` and pass it to BigDecimal like we did. I couldn't think of a way to duck(type) myself out of this one 🤔